wallet-new/src/Cardano/Wallet/Util.hs

```diff
@@ -23,3 +29,26 @@ stripFieldPrefix = dropWhile (not . isUpper)
 
 mkJsonKey :: String -> String
 mkJsonKey s = fromMaybe s . headToLower $ stripFieldPrefix s
+
+-- * Time
+
+-- | Currently we support only American usage.
+defaultApiTimeLocale :: Time.TimeLocale
+defaultApiTimeLocale = Time.defaultTimeLocale
+
+-- | Time format used in API. Corresponds to ISO-8601.
+-- Example of time: "2018-03-07T16:20:27.477318"
+--
+-- Note: there is more neat support of time formats in time-1.9,
+-- Data.Time.Format.ISO8601 module, but that version is barely applicable with
+-- current LTS-9.1.
+apiTimeFormat :: String
+apiTimeFormat = Time.iso8601DateFormat (Just "%H:%M:%S%Q")
+
+-- | Parse UTC time from API.
+parseApiUtcTime :: MonadFail m => Text -> m Time.UTCTime
+parseApiUtcTime = Time.parseTimeM False defaultApiTimeLocale apiTimeFormat . toString
+
+-- | Encode UTC time for API.
+showApiUtcTime :: Time.UTCTime -> Text
```

Comment:
That's probably trivial, but if we have time it would be nice to write a QuickCheck property to assess that `showApiUtcTime` and `parseApiUtcTime` roundtrip.

Reply:
Right, that is must-have, added.